packages/core/addon/chart-builders/base.ts

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright 2020, Yahoo Holdings Inc.
+ * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
+ */
+import Mixin from '@ember/object/mixin';
+import EmberObject from '@ember/object';
+import { ResponseV1 } from 'navi-data/serializers/facts/interface';
+import RequestFragment from 'navi-core/models/bard-request-v2/request';
+
+type ResponseRow = ResponseV1['rows'][number];
+
+export default abstract class BaseChartBuilder extends EmberObject {
+  /**
+   * @param row - single row of fact data
+   * @returns name of x value given row belongs to
+   */
+  abstract getXValue(row: ResponseRow, config: unknown, request: RequestFragment): string;
+
+  /**
+   * @param response - response from fact service
+   * @param config
+   * @param config.metrics - list of metrics to chart
+   * @param request - request used to get data
+   * @returns array of c3 data with x values
+   */
+  abstract buildData(response: ResponseV1, _config: unknown, request: RequestFragment): unknown[];
```

Comment:
```suggestion
  abstract buildData(response: ResponseV1, _config: unknown, request: RequestFragment): C3Row[];
```

and add 
https://github.com/yahoo/navi/blob/cba1907a9f85ad96e8fe5edc9246e35ac83970f0/packages/core/addon/chart-builders/base.ts#L12-L17
(slightly annoying thing about this type is that when creating a C3Row you have to explicitly cast `as C3Row` because `x` has a different shape compared to the other keys)